projects/lib/src/lib/index.ts

```diff
@@ -212,7 +212,7 @@ export const syncStateUpdate = (
             } catch (e) {
                 console.warn('Unable to save state to localStorage:', e);
             }
-        } else if (typeof stateSlice === 'undefined' && removeOnUndefined) {
+        } else if (typeof stateSlice === 'undefined' && removeOnUndefined && storage !== undefined) {
             try {
                 storage.removeItem(storageKeySerializer(key as string));
             } catch (e) {
```

Comment:
@aanaya-deltacom thanks for the PR.

Is it necessary to add that check, given we are catching any errors here? So if storage is undefined, it should catch that error and add a console warning.

Reply:
Yes, I think that it is necessary to be consistent with the expected behavior using `checkStorageAvailability: true` config option.

For example, just before ([L200](https://github.com/btroncone/ngrx-store-localstorage/blob/master/projects/lib/src/lib/index.ts#L200)) it is being checked in order to avoid the warning when saving in the local storage, so I think that the same behavior is expected when removing.

Reply:
Ok makes sense. Thanks for clarifying.